tests/neuro_san/internals/graph/activations/test_abstract_class_activation.py

```diff
@@ -0,0 +1,406 @@
+        assert activation_instance.arguments["test_arg"] == "test_value"
+        assert activation_instance.arguments.get("origin") is not None
+        assert activation_instance.arguments.get("origin_str") == "test_full_name"
+        assert activation_instance.arguments.get("progress_reporter") is not None
+
+    def test_get_full_class_ref(self, activation_instance):
+        """Test that get_full_class_ref returns the correct class reference."""
+        assert activation_instance.get_full_class_ref() == "test_module.TestClass"
+
+    @pytest.mark.asyncio
+    async def test_build_success(self, activation_instance):
+        """Test successful build with a valid CodedTool."""
+        mock_tool = MockCodedTool()
+
+        with patch.object(activation_instance, 'resolve_class', return_value=MockCodedTool):
+            with patch.object(activation_instance, 'instantiate_coded_tool', return_value=mock_tool):
+                with patch.object(activation_instance, 'attempt_invoke', new_callable=AsyncMock,
+                                  return_value="test_result"):
+                    result = await activation_instance.build()
+
+                    assert isinstance(result, AIMessage)
+                    assert result.content == "test_result"
+
+    @pytest.mark.asyncio
+    async def test_build_with_non_coded_tool(self, activation_instance):
+        """Test build when instantiated object is not a CodedTool."""
+        class NotACodedTool:
+            """A class that does not inherit from CodedTool."""
+
+        with patch.object(activation_instance, 'resolve_class', return_value=NotACodedTool):
+            with patch.object(activation_instance, 'instantiate_coded_tool', return_value=NotACodedTool()):
+                result = await activation_instance.build()
+
+                assert isinstance(result, AIMessage)
+                assert "Error:" in result.content
+                assert "is not a CodedTool" in result.content
+
+    def test_resolve_class_first_level_success(self, activation_instance):
+        """Test resolving class at the most specific level."""
+        mock_resolver = MagicMock()
+        mock_resolver.resolve_class_in_module.return_value = MockCodedTool
+
+        with patch(RESOLVER_PATH, return_value=mock_resolver):
+            result = activation_instance.resolve_class("TestClass", "test_module")
+
+            assert result == MockCodedTool
+            # Should succeed on first attempt
+            assert mock_resolver.resolve_class_in_module.call_count == 1
+
+    def test_resolve_class_second_level_success(self, activation_instance):
+        """Test resolving class after failing at first level."""
+        mock_resolver_fail = MagicMock()
+        mock_resolver_fail.resolve_class_in_module.side_effect = ValueError("Not found")
+
+        mock_resolver_success = MagicMock()
+        mock_resolver_success.resolve_class_in_module.return_value = MockCodedTool
```

Comment:
Could use the `tests.fixtures` folder with real paths and classes to rely less on Mocks and simplify the test.